server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -1015,7 +1031,7 @@ public byte id() {
 
         @Override
         public Prepared prepare(long minUtcMillis, long maxUtcMillis) {
-            return wrapPreparedRounding(delegate.prepare(minUtcMillis, maxUtcMillis));
+            return wrapPreparedRounding(delegate.prepare(minUtcMillis - offset, maxUtcMillis - offset));
```

Comment:
is this change fixing an existing bug?

Reply:
Yes, but I think the bug is worse with this change.

Reply:
The new assertions in the `ArrayRounding` fail without this.